### PR TITLE
Enable configuration of verification suites and tests

### PR DESCRIPTION
**Pull request by arothuis**
[Ticket 104934296](https://www.pivotaltracker.com/story/show/104934296)

### CODE REVIEW

**Comment by arothuis** [comment]
@rjkip @DRvanR :money_with_wings:

**Comment by DRvanR** [comment]
Dont make it parameters

File: composer.json [hunk 1]
```diff
@@ -30,7 +30,8 @@
   "require-dev": {
     "sensio/generator-bundle": "~2.3",
     "ibuildings/qa-tools": "^1.1",
-    "matthiasnoback/symfony-config-test": "^1.3"
+    "matthiasnoback/symfony-config-test": "^1.3",
+    "matthiasnoback/symfony-dependency-injection-test": "^0.7.4"
```
Comment (rjkip) [inline composer.json]: Do you still use this?

File: src/Surfnet/Conext/OperationsSupportBundle/Tests/DependencyInjection/ConfigurationTest.php [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                    'test1',
+                    'test2'
+                ],
+                'suite_name_2' => [
+                    'test1'
+                ]
+            ]
+        ] );
+
+        $this->assertConfigurationIsValid([$config]);
```
Comment (rjkip) [inline src/Surfnet/Conext/OperationsSupportBundle/Tests/DependencyInjection/ConfigurationTest.php]: What does this test?

**Comment by arothuis** [comment]
@rjkip @DRvanR updated

File: src/Surfnet/Conext/OperationsSupportBundle/Tests/DependencyInjection/ConfigurationTest.php [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+        $this->assertConfigurationIsInvalid([$config], 'Expected array');
+    }
+
+    /**
+     * @test
+     * @group EntityVerificationFramework
+     * @group Configuration
+     */
+    public function parses_configuration_as_multi_array()
```
Comment (rjkip) [inline src/Surfnet/Conext/OperationsSupportBundle/Tests/DependencyInjection/ConfigurationTest.php]: `can_contain_multiple_suites`?

**Comment by rjkip** [comment]
Bar the test name, :+1:

File: src/Surfnet/Conext/OperationsSupportBundle/Tests/DependencyInjection/ConfigurationTest.php (hunk 3 above)
Comment (arothuis) [inline src/Surfnet/Conext/OperationsSupportBundle/Tests/DependencyInjection/ConfigurationTest.php]: I agree.

File: app/config/config.yml [hunk 4]
```diff
@@ -60,3 +60,11 @@ surfnet_janus_api_client:
     base_url: %janus_api_base_url%
     username: %janus_api_username%
     password: %janus_api_password%
+
+surfnet_conext_operations_support:
+    suites:
+        example:
+            - example_test
+        some_suite:
+            - my_custom_verification
```
Comment (DRvanR) [inline app/config/config.yml]: please remove default suites and tests that do not exist.

File: app/config/config_test.yml [hunk 5]
```diff
@@ -17,3 +17,6 @@ monolog:
 
 swiftmailer:
     disable_delivery: true
+
+surfnet_conext_operations_support:
+    suites: []
```
Comment (DRvanR) [inline app/config/config_test.yml]: This should be prod config, test should not differ (hence the reason to not have an invalid config in the base config)

File: src/Surfnet/Conext/OperationsSupportBundle/DependencyInjection/SurfnetConextOperationsSupportExtension.php [hunk 6]
```diff
@@ -18,22 +18,39 @@
...
+                $testClass = NameResolver::resolveToClass("$suiteName.$testName");
+                $testDefinition = new Definition($testClass);
+
+                $suiteDefinition->addMethodCall('addVerificationTest', [$testDefinition]);
+            }
+
+            $runner->addMethodCall('addVerificationSuite', [$suiteDefinition]);
+        }
+
+        $container->setDefinition('surfnet_conext_operations_support.verification_runner', $runner);
```
Comment (DRvanR) [inline src/Surfnet/Conext/OperationsSupportBundle/DependencyInjection/SurfnetConextOperationsSupportExtension.php]: why is the runner definition overwritten by the exact same definition?

**Comment by arothuis** [comment]
:poodle: @DRvanR @rjkip

**Comment by rjkip** [comment]
:+1:

**Comment by DRvanR** [comment]
:+1:

**Pull request merged**